src/app/view/components/main-legacy/main-legacy.component.ts

```diff
@@ -41,12 +50,24 @@ export class MainComponentLegacyComponent implements OnInit {
     ngOnInit(): void {
         document.body.style.backgroundColor = '';
         document.body.style.backgroundImage = '';
-        this.router.events.subscribe((_event) => this.setMenuState());
+        this.router.events.pipe(filter((event) => event instanceof NavigationEnd)).subscribe((_event) => this.setMenuState());
```

Comment:
I don't understand why this code is here... I didn't want it to touch it much, but I just make the callback happen on ```NavigationEnd```, need to deeper check this.